pkg/render/render.go

```diff
@@ -120,21 +124,41 @@ func Calico(
 			825*24*time.Hour, // 825days*24hours: Create cert with a max expiration that macOS 10.15 will accept
 			nil,
 			ManagerServiceIP,
-			ManagerServiceDNS,
 		)
 		if err != nil {
 			return nil, fmt.Errorf("generating certificates for manager was not finalized due to %v", err)
 		}
 		tss = append(tss, managerTLSSecret)
 	}
 
+	// In a Calico Enterprise management cluster we need a secondary TLS certificate for internal communication
+	// between voltron and other components (e.g. kube controller and compliance server).
+	var voltronTLSSecret *corev1.Secret
+	if isManagement {
```

Comment:
I think this is the only place we need this generation because we only need Kube controller and compliance server to have this cert set when it's a management cluster. 

Looks like both of those components treat this as optional setting so it should support this only being available for management cluster type.

Reply:
I also don't check that it's empty before doing this, because this `voltron-tls` secret is not intended to be overridden by the user.